relayer/crates/starknet-chain-components/src/impls/messages/create_client.rs

```diff
@@ -73,10 +74,21 @@ where
             chain_id: payload.client_state.chain_id,
         };
 
+        // Convert Vec<u8> to a slice of 8 u32 values.
+        let mut root_u32: Vec<u32> = Vec::new();
```

Comment:
actually you don't even have to maintain this `root_u32: Vec`.

```rs
if root.len() % 4 != 0 {
  return Chain::raise_error(SOME_ERROR_MESSAGE);
}

let mut root_slice: [u32; 8] = [0; 8];

for (slice_u32, chunk) in root_slice.zip(root.chunks(4)) {
  *slice_u32 = u32::from_le_bytes([chunk[0], chunk[1], chunk[2], chunk[3]])
}
```

Reply:
make sure `root.len()` is exactly 256 -- otherwise, it will not fill the `root_slice` correctly.

Reply:
Done in ee085c68f267ab8c7fce9b099ef4da7084a9fd64